### DISCUSSION TITLE
Filament engine draws straight lines and how to adjust line width

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#6928] [General]
In Opengl, there is an API called glLineWidth that can set different line widths. Which API is used in the filling engine

thanks

**Comment by bejado** [comment]
We don't provide this functionality directly in Filament. To draw a straight line, you can create quad geometry and position the vertices accordingly to achieve the desired width.

**Comment by bejado** [comment]
We don't provide this functionality directly in Filament. To draw a straight line, you can create quad geometry and position the vertices accordingly to achieve the desired width.

**Comment by Haidiye00** [comment]
Will an API like glLineWidth be added in the future
